inductiva/molecules/scenarios/protein_solvation/post_processing.py

```diff
@@ -3,56 +3,110 @@
+        else:
+            self.trajectory = os.path.join(self.sim_output_dir,
+                                           "solvated_protein.trr")
+        self.universe = unwrap_trajectory(self.topology, self.trajectory)
 
     def render_interactive(self,
                            representation: Literal["cartoon", "ball+stick",
                                                    "line", "point",
                                                    "ribbon"] = "ball+stick",
-                           add_backbone: bool = True,
-                           use_compressed_trajectory: bool = False):
+                           add_backbone: bool = True):
         """Render the simulation outputs in an interactive visualization.
         Args: 
             representation: The protein representation to use for the 
             visualization.
             add_backbone: Whether to add the protein backbone to the 
             visualization.
-            use_compressed_trajectory: Whether to use the compressed trajectory. 
             """
-        topology = os.path.join(self.sim_output_dir, "solvated_protein.tpr")
-        if use_compressed_trajectory:
-            trajectory = os.path.join(self.sim_output_dir, "trajectory.xtc")
-        else:
-            trajectory = os.path.join(self.sim_output_dir,
-                                      "solvated_protein.trr")
-
-        universe = unwrap_trajectory(topology, trajectory)
-        view = nv.show_mdanalysis(universe)
+        view = nv.show_mdanalysis(self.universe)
         view.add_representation(representation,
                                 selection="not water and not ion")
         if add_backbone:
             view.add_representation("cartoon", selection="protein")
         print("System Information:")
-        print(f"Number of atoms in the system: {len(universe.atoms)}")
+        print(f"Number of atoms in the system: {len(self.universe.atoms)}")
         print(f"Number of amino acids:"
-              f"{universe.select_atoms('protein').n_residues}")
+              f"{self.universe.select_atoms('protein').n_residues}")
         print(f"Number of solvent molecules:"
-              f"{universe.select_atoms('not protein').n_residues}")
-        print(f"Number of trajectory frames: {len(universe.trajectory)}")
+              f"{self.universe.select_atoms('not protein').n_residues}")
+        print(f"Number of trajectory frames: {len(self.universe.trajectory)}")
         return view
+
+    def calculate_rmsf_trajectory(self, nglview_visualization=True):
+        """Calculate the root mean square fluctuation (RMSF) of the protein 
+        residues over the trajectory. It is typically calculated for the C&alpha 
+        atom of each residue. It is the square root of the variance of 
+        the fluctuation around the average position:
+        &rhoi = √⟨(xi - ⟨xi⟩)²⟩
+        It quantifies how much a structure diverges from a reference over time, 
```

Comment:
What is the reference? Is the initial protein state?

Reply:
No, in my implementation is the average structure calculated over the whole simulation.

Reply:
No, in my implementation I calculated the average structure in the whole simulation.

Reply:
I would add that when you say a reference.